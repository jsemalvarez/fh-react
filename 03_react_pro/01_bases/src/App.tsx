import './App.css'
import { Counter } from './bases/Counter'
import { CounterBy } from './bases/CounterBy'
import { CounterEffect } from './bases/CounterEffect'
import { CounterReducerComponent } from './bases/CounterReducer'

function App() {

  return (
    <>
      <h1>React Bases</h1>
      <hr />

      <Counter initialValue={ 15 }/>
      <hr />

      <CounterBy />
      <hr />

      <CounterEffect />
      <hr />

      <CounterReducerComponent />
    </>
  )
}

export default App
