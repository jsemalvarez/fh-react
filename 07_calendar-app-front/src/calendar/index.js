export * from './page/CalendarPage'
export * from './components/Navbar'