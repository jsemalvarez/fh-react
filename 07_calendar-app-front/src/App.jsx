import { BrowserRouter } from 'react-router-dom'
import './App.css'
import { AppRouter } from './router'
import { Provider } from 'react-redux'
import { store } from './store'

export const App = () => {
  return (
    <Provider store={ store }>
      <BrowserRouter>
        <AppRouter />
      </BrowserRouter>
    </Provider>
  )
}
