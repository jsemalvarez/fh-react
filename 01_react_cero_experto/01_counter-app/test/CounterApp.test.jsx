import { fireEvent, render, screen } from "@testing-library/react"
import { CounterApp } from "../src/CounterApp"

describe('CounterApp', () => {

    const initlaValue = 10;

    test('should be to match with the snapshot',()=> {

        const { container } = render( <CounterApp value={ initlaValue } /> );

        expect( container ).toMatchSnapshot();
    })

    test('should display value = 10', () => {

        render( <CounterApp value={ initlaValue } /> ); 
        expect( screen.getByText(10) ).toBeTruthy();
    })

    test('should increment by one ', () => {

        render( <CounterApp value={initlaValue}/> );
        fireEvent.click( screen.getByText('+1') );

        expect( screen.getByText('11')).toBeTruthy();
    })

    test('should subtract one ', () => {

        render( <CounterApp value={initlaValue}/> );
        fireEvent.click( screen.getByText('-1') );

        expect( screen.getByText('9')).toBeTruthy();
    })

    test('should reset the counter ', () => {

        render( <CounterApp value={ 333 }/> );
        fireEvent.click( screen.getByText('+1') );
        fireEvent.click( screen.getByText('+1') );
        fireEvent.click( screen.getByText('+1') );

        // fireEvent.click( screen.getByText('Reset') );
        fireEvent.click(screen.getByRole('button', { name: 'btn-reset' }));

        expect( screen.getByText('333')).toBeTruthy();
    })
})