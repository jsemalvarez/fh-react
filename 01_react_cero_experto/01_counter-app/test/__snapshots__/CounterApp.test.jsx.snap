// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CounterApp should be to match with the snapshot 1`] = `
<div>
  <h1>
    CounterApp
  </h1>
  <h2>
     
    10
     
  </h2>
  <button>
     +1 
  </button>
  <button>
     -1 
  </button>
  <button
    aria-label="btn-reset"
  >
     Reset 
  </button>
</div>
`;
